ArduCopter/switches.cpp

```diff
@@ -624,6 +625,38 @@ void Copter::do_aux_switch_function(int8_t ch_function, uint8_t ch_flag)
             }
 #endif
             break;
+
+        case AUXSW_WINCH_ENABLE:
+            switch (ch_flag) {
+                case AUX_SWITCH_HIGH:
+                    // high switch maintains current position
+                    g2.winch.release_length(0.0f);
+                    Log_Write_Event(DATA_WINCH_LENGTH_CONTROL);
```

Comment:
I have nothing against writing these events, but the AUXSW_WINCH_CONTROL will override this and have no event written.

Reply:
I don't quite understand why these events wouldn't be written..

Reply:
These will be written. What I'm saying is that in AUXSW_WINCH_CONTROL it doesn't write any event. So even when an user uses the rate control with an aux switch, the log can still show length control as the latest event.

Reply:
OK, I think I understand.  So the issue is that if the ch6 tuning knob is used (i.e see tuning.cpp) then there are no DATA_WINCH_RATE_CONTROL messages written to the log files and also the ch6 control will conflict with the aux switch controls if they are setup (the winch would momentarily try to do length control if the aux switches were changed but would then immediately revert to rate control).  I think we can handle this with some documentation on the still-to-be-created wiki page for the winch. I don't think the missing event messages for the rate-control while using ch6 tuning are too important.